### PR TITLE
feat: `avm/res/aad/domain-service`

### PR DESCRIPTION
**Pull request by ReneHezser**
## Description

Adds Azure Active Directory Domain Services module. Additional changes:
- The module requires changes to the testing process, as only one instance can be provisioned in a tenant.
- The script generating certificates for testing has been fixed.

## Pipeline Reference

| Pipeline |
| -------- |
|   [avm.res.aad.domain-service](https://github.com/ReneHezser/bicep-registry-modules/actions/workflows/avm.res.aad.domain-service.yml)       |

## Type of Change

- [x] Update to CI Environment or utlities (Non-module effecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [x] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: .github/CODEOWNERS
Comment (eriqua) [inline .github/CODEOWNERS:6]: Placeholder until team membership is resolved

Comment (eriqua) [reply .github/CODEOWNERS:6]: Hi @ReneHezser, please don't resolve the comment until the codeowners file shows no errors.
Relevant azure teams must be created and correct membership must be set up for you to be able to approve future contributions to this module. Ref https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr20---category-contributionsupport---github-teams-only

Please don't hesitate to reach out if any support needed!

Comment (ReneHezser) [reply .github/CODEOWNERS:6]: Sorry. I'll wait for you to create the necessary groups.

Comment (eriqua) [reply .github/CODEOWNERS:6]: Hi @ReneHezser creating contributor and owner groups is part of the module owners' responsabilities. The link above is providing guidelines on how to do that, but in brief you should:
- Create a `avm-res-aad-domainservice-module-owners-bicep` team in the Azure organization
- Add other module owners as team members
- Request the team to be a member of the parent team `avm-technical-reviewers-bicep`
- Wait for someone in the core team to approve the membership
- Create a `avm-res-aad-domainservice-module-contributors-bicep` team in the Azure organization
- Add module contributors as team members
- Request the team to be a member of the parent team `avm-module-contributors-bicep`
- Wait for someone in the core team to approve the membership

**Comment by ChrisSidebotham** [comment]
@ReneHezser - Great work on this, @Azure/avm-core-team-technical-bicep will need to update the existing AVM Test environment before merging with the criteria set in the Notes section of your ReadMe :)

**Comment by ReneHezser** [comment]
[heart]         René Hézser reacted to your message:
________________________________

File: .github/CODEOWNERS
Comment (eriqua) [reply .github/CODEOWNERS:6]: Thank you @ReneHezser! The membership is now fixed and the codeowners file is valid. Resolving

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1
Comment (eriqua) [inline avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:115]: Just earlier today, we merged a PR enforcing PowerShell formatting that should avoid these changes. Could you please merge latest back to you fork branch, add a space anywhere to this file and save? That should enforce the new aligned formatting to this file. Thanks!

File: avm/res/aad/domain-service/tests/e2e/waf-aligned/dependencies.bicep
Comment (AlexanderSehr) [inline avm/res/aad/domain-service/tests/e2e/waf-aligned/dependencies.bicep:157]: I'm curious, why not use:
```suggestion
output subnetResourceId string = virtualNetwork.properties.subnets[1].id
```

File: avm/utilities/e2e-template-assets/scripts/Set-PfxCertificateInKeyVault.ps1
Comment (AlexanderSehr) [inline avm/utilities/e2e-template-assets/scripts/Set-PfxCertificateInKeyVault.ps1:15]: Could it be the case that the latter half of the sentence was cut?

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1
Comment (AlexanderSehr) [inline avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:208]:
```suggestion
        Set-ItResult -Skipped -Because "only one instance of the Domain-Service can be deployed at a time, and as such, also only one test can exist at a time."
```

File: avm/res/aad/domain-service/main.bicep
Comment (AlexanderSehr) [inline avm/res/aad/domain-service/main.bicep:227]: Not using the latest schema. Please update as per [specs](https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#diagnostic-settings), esepecially the 'logs'.

File: avm/res/aad/domain-service/tests/e2e/waf-aligned/main.test.bicep
Comment (AlexanderSehr) [inline avm/res/aad/domain-service/tests/e2e/waf-aligned/main.test.bicep:22]:
```suggestion
```

Should not be part of the tests

File: avm/res/aad/domain-service/tests/e2e/waf-aligned/main.test.bicep
Comment (AlexanderSehr) [inline avm/res/aad/domain-service/tests/e2e/waf-aligned/main.test.bicep:26]: I know why you used `vars` here, but please note - any element that is not part of the 

```bicep
module testDeployment '../../../main.bicep' = {
  scope: resourceGroup
  (...)
}
```
block will not find its way into the readme `usageExamples`.

Duplicatting the configuration in both the init & idem may seem, odd, yet I'd recommend to actually do that instead of using vars for the above reason.

Thoughts?

File: avm/res/aad/domain-service/tests/e2e/waf-aligned/main.test.bicep
Comment (AlexanderSehr) [inline avm/res/aad/domain-service/tests/e2e/waf-aligned/main.test.bicep:93]: I'm a bit blown away - does that work nowadays? I recall, that originally, you could not reference an `existing` resource if set resource did not exist at the start of the deployment (which in this case it definitely doesn't as it is deployed via the dependencies).

File: avm/res/aad/domain-service/tests/e2e/waf-aligned/main.test.bicep
Comment (ReneHezser) [reply avm/res/aad/domain-service/tests/e2e/waf-aligned/main.test.bicep:26]: That makes sense. Duplicating them again.

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1
Comment (AlexanderSehr) [inline avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:333]: Hey @ReneHezser why this update? The script contributors ideally run is the `Set-AVMModule` script which is directly in the tools folder as opposed to the one nested in the pipelines folder.

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1
Comment (AlexanderSehr) [inline avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:1160]: I assume this was unintentional?

File: avm/res/aad/domain-service/tests/e2e/waf-aligned/main.test.bicep
Comment (AlexanderSehr) [inline avm/res/aad/domain-service/tests/e2e/waf-aligned/main.test.bicep:84]: Question: How come you disabled any log collection? Just curious 😄

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1
Comment (ReneHezser) [reply avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:333]: I did not modify this on purpose.

File: avm/res/aad/domain-service/tests/e2e/waf-aligned/main.test.bicep
Comment (ReneHezser) [reply avm/res/aad/domain-service/tests/e2e/waf-aligned/main.test.bicep:84]: Copy, paste, don't think ¯\_(ツ)_/¯

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1
Comment (AlexanderSehr) [inline avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:1163]: Seems to be one more leftover, then we are good 💪

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**
